Dalamud/Plugin/Services/IBuddyList.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+using Dalamud.Game.ClientState.Buddy;
+
+namespace Dalamud.Plugin.Services;
+
+/// <summary>
+/// This collection represents the buddies present in your squadron or trust party.
+/// It does not include the local player.
+/// </summary>
+public interface IBuddyList : IReadOnlyCollection<BuddyMember>
+{
+    /// <summary>
+    /// Gets the amount of battle buddies the local player has.
+    /// </summary>
+    public int Length { get; }
+    
+    /// <summary>
+    /// Gets a value indicating whether the local player's companion is present.
+    /// </summary>
+    [MemberNotNullWhen(true, "CompanionBuddy")]
+    public bool CompanionBuddyPresent { get; }
+    
+    /// <summary>
+    /// Gets a value indicating whether the local player's pet is present.
+    /// </summary>
+    [MemberNotNullWhen(true, "PetBuddy")]
+    public bool PetBuddyPresent { get; }
+    
```

Comment:
Unsure if either of these should be part of the public API, given that they're just checking the relevant `Buddy` for null already.

Reply:
That's fair, in normal use I would more likely just check for null like this:
![image](https://github.com/goatcorp/Dalamud/assets/9083275/b427d631-c8fd-49c6-8cce-a34d9adde275)

Reply:
Hm. Unless there's a way for these objects to be not-null but still not be "present" (previously spawned but no longer around?), let's remove these from the interface and mark them deprecated in the base service.